include/tdi/common/tdi_table_key.hpp

```diff
@@ -66,30 +90,71 @@ class KeyFieldValueTernary : public KeyFieldValue {
  public:
   KeyFieldValueTernary(T &value, T &mask)
       : value_(value), mask_(mask){};
-  KeyFieldValueTernary(uint8_t *value_ptr,
-                       uint8_t *mask_ptr,
-                       size_t &size)
+  KeyFieldValueTernary(T *value_ptr,
+                       T *mask_ptr,
```

Comment:
This is wrong. It was intentionally kept as uint8_t which is bytearray. It is not to be made into a template